src/main/java/org/ibp/api/java/impl/middleware/study/StudyEntryServiceImpl.java

```diff
@@ -91,13 +116,45 @@ public StudyEntryDto replaceStudyEntry(final Integer studyId, final Integer entr
 			.replaceStudyEntry(studyId, entryId, gid, this.pedigreeService.getCrossExpansion(gid, this.crossExpansionProperties));
 	}
 
+	@Override
+	public List<StudyEntryDto> createStudyEntries(final Integer studyId,
+		final StudyEntryGeneratorBatchRequestDto studyEntryGeneratorBatchRequestDto) {
+		this.studyValidator.validate(studyId, true);
+		if(studyEntryGeneratorBatchRequestDto.getListId() != null && studyEntryGeneratorBatchRequestDto.getListId() != 0) {
```

Comment:
@darla-leafnode @mderamos In my opinion this is not clear for external clients using the service. 
If I see a service where I can send a listId but also an entryTypeId I would expect the app to create entries with the entry type I specified and If I am not mistaken we are just checking if listId is specified and creating the stocks with ENTRY type. 
I would suggest to review this a little bit more and either add proper validations and prevent users to think that they can send any type combined with a listId or reorganize a bit better the RequestDTO

Reply:
Hi @clarysabel, good point. I just created a new endpoint and requestdto with listid as attribute to removed the confusion.